engine/grid.py

```diff
@@ -81,21 +97,20 @@ def generate_nodes(start_lat, start_long, rows, cols, step_size_m):
             # traversal.
             for i in range(cols):
                 for j in range(rows):
-                    is_border = (j == 0 or j == rows - 1)
```

Comment:
I feel like I remember another comment about test case failing because is_border isn't initialize. I'm not sure if the new code is pushed but when I tested it, first of all, I couldn't pull because there was a bunch of merge conflicts, but if I don't pull and run the test, it crashed and said this wasn't initialized

Reply:
If the code was pushed along with the merging main step, make sure to keep commits as separate as possible

Reply:
I made the change u mentioned in commit https://github.com/cornellnexus/src/pull/108/commits/4137dcf81c4ab1d798c42a64afc67c962176c27b bc we defined what a border node was differently before so i moved the is_border node definition to get_all_lawnmower

Reply:
a bit confused about what wasn't working?

Reply:
Merge conflict when pulling